vortex-array/src/array/chunked/canonical.rs

```diff
@@ -140,21 +138,15 @@ fn swizzle_struct_chunks(
     validity: Validity,
     struct_dtype: &StructDType,
 ) -> VortexResult<StructArray> {
-    let chunks: Vec<StructArray> = chunks.iter().map(StructArray::try_from).try_collect()?;
-
     let len = chunks.iter().map(|chunk| chunk.len()).sum();
-
     let mut field_arrays = Vec::new();
 
     for (field_idx, field_dtype) in struct_dtype.dtypes().iter().enumerate() {
-        let mut field_chunks = Vec::new();
-        for chunk in &chunks {
-            field_chunks.push(
-                chunk
-                    .field(field_idx)
-                    .ok_or_else(|| vortex_err!("All chunks must have same dtype; missing field at index {}, current chunk dtype: {}", field_idx, chunk.dtype()))?,
-            );
-        }
```

Comment:
I think we are going to have more of these - you can't assume encoding given the dtype